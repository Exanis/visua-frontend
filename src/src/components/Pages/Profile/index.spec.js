import React from 'react';
import {shallow} from 'enzyme';

import {ProfilePage} from './index';

describe('<ProfilePage />', () => {
    const defaultProps = {
        user: {
            update: {
                haveResult: false,
                success: false
            },
            info: {}
        },
        classes: {},
        userActions: {
            updateUser: () => {
            },
        },
        intl: {
            formatMessage: () => {}
        }
    };


    it('should render ProfilePage properly', () => {
        const rendererComponent = shallow(<ProfilePage {...defaultProps} />);

        expect(rendererComponent).toMatchSnapshot();
    });
});