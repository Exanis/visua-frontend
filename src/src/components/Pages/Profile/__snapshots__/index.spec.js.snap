// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> should render ProfilePage properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfilePage
    classes={Object {}}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    user={
        Object {
            "info": Object {},
            "update": Object {
              "haveResult": false,
              "success": false,
            },
          }
    }
    userActions={
        Object {
            "updateUser": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <VisuaSnackbar
          open={false}
          variant="error"
>
          <FormattedMessage
                    defaultMessage="Error while updating your profile; please check the form."
                    id="profile.update.error"
                    values={Object {}}
          />
</VisuaSnackbar>,
        <VisuaSnackbar
          open={false}
          variant="success"
>
          <FormattedMessage
                    defaultMessage="Profile updated."
                    id="profile.update.success"
                    values={Object {}}
          />
</VisuaSnackbar>,
        <WithStyles(Typography)
          variant="title"
>
          <FormattedMessage
                    defaultMessage="My profile"
                    id="profile.title"
                    values={Object {}}
          />
</WithStyles(Typography)>,
        <form
          className={undefined}
>
          <WithStyles(Typography)
                    variant="caption"
          >
                    <FormattedMessage
                              defaultMessage="Username"
                              id="profile.username"
                              values={Object {}}
                    />
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    variant="body1"
          />
          <br />
          <TextField
                    error={false}
                    label={undefined}
                    onChange={[Function]}
                    required={false}
                    select={false}
                    value={undefined}
          />
          <TextField
                    error={false}
                    label={undefined}
                    onChange={[Function]}
                    required={false}
                    select={false}
                    value={undefined}
          />
          <TextField
                    error={false}
                    label={undefined}
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="email"
                    value={undefined}
          />
          <br />
          <WithStyles(Typography)
                    variant="caption"
          >
                    <FormattedMessage
                              defaultMessage="To keep your current password, leave those fields empty."
                              id="profile.passwordNotice"
                              values={Object {}}
                    />
          </WithStyles(Typography)>
          <TextField
                    error={false}
                    label={undefined}
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="password"
                    value=""
          />
          <TextField
                    error={false}
                    label={undefined}
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="password"
                    value=""
          />
          <br />
          <WithStyles(Button)
                    onClick={[Function]}
                    variant="contained"
          >
                    <FormattedMessage
                              defaultMessage="Update profile"
                              id="profile.update"
                              values={Object {}}
                    />
          </WithStyles(Button)>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormattedMessage
            defaultMessage="Error while updating your profile; please check the form."
            id="profile.update.error"
            values={Object {}}
/>,
          "open": false,
          "variant": "error",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMessage": "Error while updating your profile; please check the form.",
            "id": "profile.update.error",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormattedMessage
            defaultMessage="Profile updated."
            id="profile.update.success"
            values={Object {}}
/>,
          "open": false,
          "variant": "success",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMessage": "Profile updated.",
            "id": "profile.update.success",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormattedMessage
            defaultMessage="My profile"
            id="profile.title"
            values={Object {}}
/>,
          "variant": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMessage": "My profile",
            "id": "profile.title",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              variant="caption"
>
              <FormattedMessage
                            defaultMessage="Username"
                            id="profile.username"
                            values={Object {}}
              />
</WithStyles(Typography)>,
            <WithStyles(Typography)
              variant="body1"
/>,
            <br />,
            <TextField
              error={false}
              label={undefined}
              onChange={[Function]}
              required={false}
              select={false}
              value={undefined}
/>,
            <TextField
              error={false}
              label={undefined}
              onChange={[Function]}
              required={false}
              select={false}
              value={undefined}
/>,
            <TextField
              error={false}
              label={undefined}
              onChange={[Function]}
              required={false}
              select={false}
              type="email"
              value={undefined}
/>,
            <br />,
            <WithStyles(Typography)
              variant="caption"
>
              <FormattedMessage
                            defaultMessage="To keep your current password, leave those fields empty."
                            id="profile.passwordNotice"
                            values={Object {}}
              />
</WithStyles(Typography)>,
            <TextField
              error={false}
              label={undefined}
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              value=""
/>,
            <TextField
              error={false}
              label={undefined}
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              value=""
/>,
            <br />,
            <WithStyles(Button)
              onClick={[Function]}
              variant="contained"
>
              <FormattedMessage
                            defaultMessage="Update profile"
                            id="profile.update"
                            values={Object {}}
              />
</WithStyles(Button)>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FormattedMessage
                defaultMessage="Username"
                id="profile.username"
                values={Object {}}
/>,
              "variant": "caption",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Username",
                "id": "profile.username",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "variant": "body1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": false,
              "label": undefined,
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": false,
              "label": undefined,
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": false,
              "label": undefined,
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "email",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FormattedMessage
                defaultMessage="To keep your current password, leave those fields empty."
                id="profile.passwordNotice"
                values={Object {}}
/>,
              "variant": "caption",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "To keep your current password, leave those fields empty.",
                "id": "profile.passwordNotice",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": false,
              "label": undefined,
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": false,
              "label": undefined,
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FormattedMessage
                defaultMessage="Update profile"
                id="profile.update"
                values={Object {}}
/>,
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Update profile",
                "id": "profile.update",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <VisuaSnackbar
            open={false}
            variant="error"
>
            <FormattedMessage
                        defaultMessage="Error while updating your profile; please check the form."
                        id="profile.update.error"
                        values={Object {}}
            />
</VisuaSnackbar>,
          <VisuaSnackbar
            open={false}
            variant="success"
>
            <FormattedMessage
                        defaultMessage="Profile updated."
                        id="profile.update.success"
                        values={Object {}}
            />
</VisuaSnackbar>,
          <WithStyles(Typography)
            variant="title"
>
            <FormattedMessage
                        defaultMessage="My profile"
                        id="profile.title"
                        values={Object {}}
            />
</WithStyles(Typography)>,
          <form
            className={undefined}
>
            <WithStyles(Typography)
                        variant="caption"
            >
                        <FormattedMessage
                                    defaultMessage="Username"
                                    id="profile.username"
                                    values={Object {}}
                        />
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        variant="body1"
            />
            <br />
            <TextField
                        error={false}
                        label={undefined}
                        onChange={[Function]}
                        required={false}
                        select={false}
                        value={undefined}
            />
            <TextField
                        error={false}
                        label={undefined}
                        onChange={[Function]}
                        required={false}
                        select={false}
                        value={undefined}
            />
            <TextField
                        error={false}
                        label={undefined}
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="email"
                        value={undefined}
            />
            <br />
            <WithStyles(Typography)
                        variant="caption"
            >
                        <FormattedMessage
                                    defaultMessage="To keep your current password, leave those fields empty."
                                    id="profile.passwordNotice"
                                    values={Object {}}
                        />
            </WithStyles(Typography)>
            <TextField
                        error={false}
                        label={undefined}
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="password"
                        value=""
            />
            <TextField
                        error={false}
                        label={undefined}
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="password"
                        value=""
            />
            <br />
            <WithStyles(Button)
                        onClick={[Function]}
                        variant="contained"
            >
                        <FormattedMessage
                                    defaultMessage="Update profile"
                                    id="profile.update"
                                    values={Object {}}
                        />
            </WithStyles(Button)>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              defaultMessage="Error while updating your profile; please check the form."
              id="profile.update.error"
              values={Object {}}
/>,
            "open": false,
            "variant": "error",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Error while updating your profile; please check the form.",
              "id": "profile.update.error",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              defaultMessage="Profile updated."
              id="profile.update.success"
              values={Object {}}
/>,
            "open": false,
            "variant": "success",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Profile updated.",
              "id": "profile.update.success",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              defaultMessage="My profile"
              id="profile.title"
              values={Object {}}
/>,
            "variant": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "My profile",
              "id": "profile.title",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="caption"
>
                <FormattedMessage
                                defaultMessage="Username"
                                id="profile.username"
                                values={Object {}}
                />
</WithStyles(Typography)>,
              <WithStyles(Typography)
                variant="body1"
/>,
              <br />,
              <TextField
                error={false}
                label={undefined}
                onChange={[Function]}
                required={false}
                select={false}
                value={undefined}
/>,
              <TextField
                error={false}
                label={undefined}
                onChange={[Function]}
                required={false}
                select={false}
                value={undefined}
/>,
              <TextField
                error={false}
                label={undefined}
                onChange={[Function]}
                required={false}
                select={false}
                type="email"
                value={undefined}
/>,
              <br />,
              <WithStyles(Typography)
                variant="caption"
>
                <FormattedMessage
                                defaultMessage="To keep your current password, leave those fields empty."
                                id="profile.passwordNotice"
                                values={Object {}}
                />
</WithStyles(Typography)>,
              <TextField
                error={false}
                label={undefined}
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                value=""
/>,
              <TextField
                error={false}
                label={undefined}
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                value=""
/>,
              <br />,
              <WithStyles(Button)
                onClick={[Function]}
                variant="contained"
>
                <FormattedMessage
                                defaultMessage="Update profile"
                                id="profile.update"
                                values={Object {}}
                />
</WithStyles(Button)>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="Username"
                  id="profile.username"
                  values={Object {}}
/>,
                "variant": "caption",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Username",
                  "id": "profile.username",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "variant": "body1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": false,
                "label": undefined,
                "onChange": [Function],
                "required": false,
                "select": false,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": false,
                "label": undefined,
                "onChange": [Function],
                "required": false,
                "select": false,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": false,
                "label": undefined,
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "email",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="To keep your current password, leave those fields empty."
                  id="profile.passwordNotice"
                  values={Object {}}
/>,
                "variant": "caption",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "To keep your current password, leave those fields empty.",
                  "id": "profile.passwordNotice",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": false,
                "label": undefined,
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": false,
                "label": undefined,
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="Update profile"
                  id="profile.update"
                  values={Object {}}
/>,
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Update profile",
                  "id": "profile.update",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
